dedupe/convenience.py

```diff
@@ -24,55 +24,94 @@ def consoleLabel(deduper): # pragma: no cover
     '''
 
     finished = False
+    has_previous = False
+    use_previous = False
     fields = unique(field.field
                     for field
                     in deduper.data_model.primary_fields)
 
 
-    while not finished :
-        n_match, n_distinct = (len(deduper.training_pairs['match']),
-                               len(deduper.training_pairs['distinct']))
-
-        uncertain_pairs = deduper.uncertainPairs()
-
+    buffer_len = 1 # Max number of previous operations
+    examples_buffer = []
+    record_pair_buffer = [] 
+
+    while not finished :     
+            
+        if has_previous and use_previous:
+            record_pair = record_pair_buffer[0]
+            record_pair_buffer = record_pair_buffer[1:]
+            examples_buffer = examples_buffer[1:]
+            use_previous = False
+        else:
+            uncertain_pairs = deduper.uncertainPairs()
```

Comment:
changed in the next commit (also fixes -i think- the potential problem of not being able to `(f)inish` if in the middle of the `for record pair in uncertain_pairs` loop)